cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -350,6 +357,28 @@ protected void configure() {
     }
   }
 
+  /**
+   * Gets all the plugins used in the program through and calls
+   * {@link #usePluginClass(String, String, String, PluginProperties, PluginSelector)} explicitly to register their
+   * usage so that they are accessible in the current configurer obtained through {@link #getConfigurer()}
+   */
+  private void registerPlugins(Set<StageSpec> stages) {
```

Comment:
I'm confused, I thought we were moving spec generation from the application to the workflow, so that it could use the workflow's configurer.

Reply:
Done.